service/task_history.go

```diff
@@ -279,41 +279,26 @@ func (uis *UIServer) taskHistoryTestNames(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	repo, err := model.FindRepository(project.Identifier)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
 	stepTime := time.Now()
-	buildVariants, err := task.FindVariantsWithTask(taskName, project.Identifier, repo.RevisionOrderNumber-50, repo.RevisionOrderNumber)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	variantsQueryDuration := time.Now().Sub(stepTime)
-	stepTime = time.Now()
-
-	taskHistoryIterator := model.NewTaskHistoryIterator(taskName, buildVariants,
+	taskHistoryIterator := model.NewTaskHistoryIterator(taskName, nil,
```

Comment:
I think you removed the variant part of the query because you didn't want to do the FindVariantsWithTask part right? I think having the variant would actually help with the other comment, and there might be a mistake with indices for the FindVariantsWithTask part. There is a build_variant_1_display_name_1_order_1 index that exists, but this index makes no sense without the project being the first key. If you add project to the beginning of this index, it'll exactly match the first stage of that query

Reply:
We can revisit this if you don't agree with the index I chose.